src/utils/node_byte_utils.ts

```diff
@@ -0,0 +1,79 @@
+import type { ByteUtils } from './byte_utils';
+
+type NodeJsEncoding = 'base64' | 'hex' | 'utf8' | 'binary';
+type NodeJsBuffer = {
+  alloc: (size: number) => Uint8Array;
+  from(array: number[]): Uint8Array;
+  from(array: Uint8Array): Uint8Array;
+  from(base64: string, encoding: NodeJsEncoding): Uint8Array;
+  isBuffer(value: unknown): value is NodeJsBuffer;
+  prototype: {
+    toString: (this: Uint8Array, encoding: NodeJsEncoding) => string;
+    equals: (this: Uint8Array, other: Uint8Array) => boolean;
+  };
+};
+
+// This can be nullish, but we gate the nodejs functions on being exported whether or not this exists
+// Node.js global
+declare const Buffer: NodeJsBuffer;
+
+function bytesToString(buffer: Uint8Array, encoding: NodeJsEncoding) {
+  return Buffer.prototype.toString.call(nodeJsByteUtils.toLocalBufferType(buffer), encoding);
+}
```

Comment:
We have to call toLocalBufferType before passing the buffer to the prototype call so wrapping this up in a helper seemed logical